File: Java/contagem.java
public class contagem {

    static int contagem(int[] dados) {
        int n = dados.length;
        int primeiro = dados[0];
        int menor = primeiro;
        int maior = n;
        if (n < primeiro) {
            menor = n;
            maior = primeiro;
        }

        int cont = 0;
        for (int i = 0; i < dados.length; i++) {
            if (dados[i] >= menor && dados[i] <= maior) {
                cont++;
            }
        }
        return cont;
    }

    public static void main(String[] args) {
        int[] dados = {3, 10, 4, 7, 2, 5, 1};
        System.out.println(contagem(dados)); // 4
    }
}

File: Js/contagem.js
function contagem(dados) {
  let n = dados.length;
  let primeiro = dados[0];
  let menor = primeiro;
  let maior = n;
  if (n < primeiro) {
    menor = n;
    maior = primeiro;
  }

  let cont = 0;
  for (let i = 0; i < dados.length; i++) {
    if (dados[i] >= menor && dados[i] <= maior) {
      cont++;
    }
  }
  return cont;
}

console.log(contagem([3, 10, 4, 7, 2, 5, 1]));